backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/UsersController.java

```diff
@@ -48,20 +55,22 @@ public UserModel getUserById(
 			.orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(id)));
 	}
 
-	//TODO: will use JSON PATCH in later PR
-	@PatchMapping({ "/{id}" })
+	@PatchMapping(path = "/{id}", consumes = "application/json-patch+json")
 	@Operation(summary = "Update a user by ID")
 	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
 	public void updateUserById(
 			@NotBlank(message = "id must not be null or blank")
 			@Parameter(description = "The id of the user.", example = "00000000-0000-0000-0000-000000000000")
 			@PathVariable String id,
-			@Validated @RequestBody UserUpdateModel userUpdateModel) {
+			@Validated @RequestBody JsonPatch patch) {
 
 		final var user = userService.getUserById(id)
 				.orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(id)));
 
-		userService.updateUser(user.getId(), userUpdateModel.getEmail());
-	}	
+		final var userPatchModel = userModelMapper.toPatchModel(user);
+		final var userPatched = jsonPatchProcessor.patch(userPatchModel, patch);
 
+		userService.updateUser(id, userModelMapper.toPatchDomain(userPatched));
```

Comment:
I wonder if it is even necessary to have a patch domain object here? Couldn't you just do something like:

``` java
final var user = userService.getUserById(id)
    .orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(id)));

final var userModel = userModelMapper.toModel(user);
final var userModelPatched = jsonPatchProcessor.patch(userModel, patch);

userService.updateUser(id, userModelMapper.toDomain(userModelPatched));
```

Reply:
Also.. to correctly handle invalid data, you can do this:

``` java
try {
    final var userModelPatched = jsonPatchProcessor.patch(userModel, patch);  
    userService.updateUser(id, userModelMapper.toDomain(userModelPatched)); 
}
catch (final ConstraintViolationException constraintViolationException) {
    final var bindException = new BindException(jsonPatch, "jsonPatch");

    constraintViolationException.getConstraintViolations().forEach(constraintViolation -> {
        final var errorCode = constraintViolation.getPropertyPath().toString();
        final var message = constraintViolation.getMessage();
        bindException.reject(errorCode, message);
    });

    throw bindException;
}
```